### PR TITLE
Allow logging in by email

### PR DESCRIPTION
**Pull request by nanaya**
To match what input box says =D

### CODE REVIEW

**Comment by peppy** [comment]
:+1:

File: app/Http/Controllers/UsersController.php [hunk 1]
```diff
@@ -75,18 +75,21 @@ public function login()
...
+            }
 
             if (Auth::check()) {
                 return Auth::user()->defaultJson();
             } else {
-                LoginAttempt::failedAttempt($ip, $username);
+                LoginAttempt::failedAttempt($ip, $usernameOrEmail);
 
-                return error_popup('wrong password or username');
+                return error_popup('wrong password or email');
```
Comment (lunafreya) [inline app/Http/Controllers/UsersController.php]: I think it needs to say "Wrong password or username/email", not "wrong password or email".

**Comment by peppy** [comment]
It does not. We are phasing out username login, so this is correct.

**Comment by avail** [comment]
no more username logins? what's the reason behind that :o

**Comment by peppy** [comment]
it's better security and also makes sure the user keeps the email up-to-date.

**Pull request merged**